driver-async/src/test/functional/com/mongodb/async/client/Fixture.java

```diff
@@ -62,9 +74,26 @@ public void apply(final SslSettings.Builder builder) {
                 }
             });
         }
+        builder.streamFactoryFactory(getStreamFactoryFactory());
         return builder;
     }
 
+    public static StreamFactoryFactory getStreamFactoryFactory() {
+        if (getSslSettings().isEnabled()) {
+            if (isNotAtLeastJava8()) {
+                return NettyStreamFactoryFactory.builder().build();
+            } else {
+                return new TlsChannelStreamFactoryFactory();
+            }
+        } else {
+            if (isNotAtLeastJava7()) {
```

Comment:
This feels strange - having one path java 7 dependent and the other java 8

Reply:
What to do though?  Is it worth re-writing TlsChannel to be Java 7-compliant, given the current low usage rate of Java 7 that we are seeing in Atlas and in public surveys?

Reply:
Yeah - roll on 4.0 :)